src/FragmentMakers.cpp

```diff
@@ -0,0 +1,80 @@
+#include "trigger/FragmentMakers.hpp"
+#include "dataformats/trigger/TriggerPrimitivesFragment.hpp"
+#include "triggeralgs/TriggerPrimitive.hpp"
+
+namespace dunedaq::trigger {
+
+//======================================================================
+std::unique_ptr<dataformats::Fragment>
+make_fragment(std::vector<TPSet>& tpsets)
+{
+  std::vector<triggeralgs::TriggerPrimitive> tps;
+  for (auto const& tpset : tpsets) {
+    for (auto const& tp : tpset.objects) {
+      tps.push_back(tp);
+    }
+  }
+
+  return make_fragment(tps);
+}
+
+//======================================================================
+std::unique_ptr<dataformats::Fragment>
+make_fragment(std::vector<triggeralgs::TriggerPrimitive>& tps)
+{
+  size_t n_bytes = sizeof(dataformats::TriggerPrimitivesFragment) +
+                   tps.size() * sizeof(dataformats::TriggerPrimitivesFragment::TriggerPrimitive);
+  std::unique_ptr<uint8_t[]> buffer(new uint8_t[n_bytes]);
+  dataformats::TriggerPrimitivesFragment* tpf = reinterpret_cast<dataformats::TriggerPrimitivesFragment*>(buffer.get());
+  tpf->version = dataformats::TriggerPrimitivesFragment::s_tpf_version;
+  tpf->n_trigger_primitives = tps.size();
+  size_t counter = 0;
+  for (auto const& tp : tps) {
+    dataformats::TriggerPrimitivesFragment::TriggerPrimitive& tp_fragment = tpf->primitives[counter++];
+    tp_fragment.time_start = tp.time_start;
+    tp_fragment.time_peak = tp.time_peak;
+    tp_fragment.time_over_threshold = tp.time_over_threshold;
+    tp_fragment.channel = tp.channel;
+    tp_fragment.adc_integral = tp.adc_integral;
+    tp_fragment.adc_peak = tp.adc_peak;
+    tp_fragment.detid = tp.detid;
+    tp_fragment.type = static_cast<uint32_t>(tp.type);
+    tp_fragment.algorithm = static_cast<uint32_t>(tp.algorithm);
+    tp_fragment.version = tp.version;
+    tp_fragment.flag = tp.flag;
```

Comment:
I have to say that I don't really like this memberwise-copy, but I don't know of a better way of doing it without introducing a dependency between at least `dataformats` and `triggeralgs`...